ig/ig.go

```diff
@@ -0,0 +1,172 @@
+package ig
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"regexp"
+	"strconv"
+)
+
+type IG struct {
+	path  string
+	image string
+	v1    int
+	v2    int
+	v3    int
+}
+
+type option func(*IG)
+
+func Path(path string) option {
+	return func(ig *IG) {
+		ig.path = path
+	}
+}
+
+func Image(image string) option {
+	return func(ig *IG) {
+		ig.image = image
+	}
+}
+
+// Runs "ig version" to get the version string
+func getIgVersionString(path string) (string, error) {
+	cmd := exec.Command(path, "version")
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	err := cmd.Run()
+	if err != nil {
+		return "", err
+	}
+	return out.String(), nil
+}
+
+// Returns the first three components of the version
+// e.g. "v0.26.0" would return (0, 26, 0)
+func extractIgVersion(str string) (int, int, int, error) {
+	versionMatcher := regexp.MustCompile(`v([0-9]+)\.([0-9]+)\.([0-9]+)`)
+	result := versionMatcher.FindStringSubmatch(str)
+	if result == nil {
+		return 0, 0, 0, fmt.Errorf("no ig version found in string: %s", str)
+	}
```

Comment:
Same as above, use github.com/blang/semver to parse it. You can see how it's used in https://github.com/inspektor-gadget/inspektor-gadget/blob/cf6328184d7eef5067f24c64047fd74411907f0f/internal/version/version.go